src/main.rs

```diff
@@ -95,14 +88,7 @@ fn run(
                 canvas.reset();
                 canvas.scale(5.0, 5.0);
 
-                let current_timestep = Instant::now();
-                lag += current_timestep.duration_since(prev_timestep).as_nanos() as f32 * 0.000001;
-                prev_timestep = current_timestep;
-
-                while lag > MS_PER_UPDATE {
-                    state = reducers::tick(&state, last_key);
-                    lag -= MS_PER_UPDATE;
-                }
+                state = reducers::tick(&state, last_key);
```

Comment:
Any chance to keep the "fixed" physics steps?

Reply:
I'm fine just closing this PR and leaving it unmerged, I just wanted to finish the game as I saw it in my head.

The idea was based upon having pacman pass exactly over each map tile once & only once, and `on_square` was the key to this: https://github.com/avantgardnerio/func_man/blob/f76a56496193da3f50edf21c5b88d1bb9d743a05/src/reducers/pacman.rs#L18

I tried to make it work on the master branch, but ran into (fixable) troubles: since he's never exactly on a square, we can trace his path, see if he crossed a square boundary, instead of treating him as exactly on a square in map space as an infinitesimal point, we can do bounds checking, and if he's gone past the boundary of a wall, we can find the intersection and round up/down accordingly, but then we need to match based upon his direction of travel, etc.

In short, I couldn't find an equally simple / terse way of doing it with the floating point position, but I'm probably over thinking it. If there is one, let me know and maybe we can unite these two branches.